### DISCUSSION TITLE
privateGPT as a system service

### DISCUSSION THREAD

**Discussion by BrunoBosshard** [imartinez/privateGPT#1330] [Q&A]
How can privateGPT be started automatically as a system service, maybe through a *.service in /etc/systemd/system?

**Comment by rmuhawieh** [comment]
> How can privateGPT be started automatically as a system service, maybe through a *.service in /etc/systemd/system?

Did you manage to figure this out?

**Reply by BrunoBosshard** [reply]
No, I am still waiting for help. Any solution also needs to consider the Python virtual environment.

**Reply by davidzmx** [reply]
did you find a solution?

**Comment by rempel1234** [comment]
This is what I wrote... may not be exactly production-ready though
[Unit]
Description=%i service with make
After=network.target

[Service]
Environment="PGPT_PROFILE=local"
User=ssm-user
Group=ssm-user
Type=forking
RemainAfterExit=true
WorkingDirectory=/home/ssm-user/private-gpt
ExecStart=/usr/bin/make run
Restart=on-failure

[Install]
WantedBy=multi-user.target
